ambry-store/src/main/java/com.github.ambry.store/StatsManager.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.config.StatsManagerConfig;
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Utils;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.io.JsonEncoder;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class StatsManager {
+  private static final Logger logger = LoggerFactory.getLogger(StatsManager.class);
+
+  private final StorageManager storageManager;
+  private final File statsOutputFile;
+  private final ScheduledExecutorService scheduler;
+  private final long publishPeriodInSecs;
+  private ScheduledFuture<?> statsAggregator = null;
+
+  /**
+   * Constructs a {@link StatsManager}.
+   * @param storageManager the {@link StorageManager} to be used to fetch the {@link Store}s.
+   * @param config the {@link StatsManagerConfig} to be used to configure the output file path and publish period.
+   * @throws IOException
+   */
+  public StatsManager(StorageManager storageManager, StatsManagerConfig config) throws IOException {
+    this.storageManager = storageManager;
+    statsOutputFile = new File(config.outputFilePath);
+    this.publishPeriodInSecs = config.publishPeriodInSecs;
+    this.scheduler = Utils.newScheduler(1, false);
+  }
+
+  /**
+   * Start the stats manager by scheduling the periodic task that collect, aggregate and publish stats.
+   */
+  public void start() {
+    statsAggregator = scheduler.scheduleAtFixedRate(new StatsAggregator(), 0, publishPeriodInSecs, TimeUnit.SECONDS);
```

Comment:
can you add some initial random delay. If not, every node during deployment will try to collect the stats at the same time.